### PR TITLE
Emulation tests

### PR DESCRIPTION
**Pull request by dlangbe**
Added rtest.py to run emulation tests with `--emulation [smoke|regression|extended]`. Tests are configured in rtest.xml and various YAML files.

Additionally added missing rank1 reduction test to BenchmarkReduction.sh.

### CODE REVIEW

File: rtest.xml [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    <emulation sets="extended">
+        <run name="extended35">./bin/rank4_reduction_test -y {REDUCTION_CONFIGS}/extended/rank4_test_params.yaml {EXTENDED_PARAMS} </run>
+    </emulation>
+    <emulation sets="extended">
+        <run name="extended36">./bin/rank5_reduction_test -y {REDUCTION_CONFIGS}/extended/rank5_test_params.yaml {EXTENDED_PARAMS} </run>
+    </emulation>
+    <emulation sets="extended">
+        <run name="extended37">./bin/rank6_reduction_test -y {REDUCTION_CONFIGS}/extended/rank6_test_params.yaml {EXTENDED_PARAMS} </run>
+    </emulation>
+</testset>
```
Comment (cgmillette) [inline rtest.xml]: newline

File: rtest.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+    parser = argparse.ArgumentParser(description="""
+    Checks build arguments
+    """)
+    parser.add_argument('-t', '--test', required=False, type=str, default=[], action='append',
+                        help='Test set to run from rtest.xml (required, e.g. osdb)')
+    parser.add_argument(      '--emulation', required=False, type=str, default=[], action='append', choices=['smoke', 'regression', 'extended'],
+                        help='Emulation set to run from rtest.xml (required, e.g. smoke)')
+    parser.add_argument(      '--name', required=False, type=str, default=[], action='append',
+                        help='Specifies tests to run from the set (optional, run all tests in the set by default)')
+    parser.add_argument('-g', '--debug', required=False, default=False,  action='store_true',
```
Comment (cgmillette) [inline rtest.py]: Are all options usable / working?

**Review by cgmillette — DISMISSED** [review]

**Review by cgmillette — DISMISSED** [review]
lgtm

**Review by cgmillette — DISMISSED** [review]
lgtm

**Review by CongMa13 — DISMISSED** [review]

**Review by cgmillette — APPROVED** [review]

**Pull request merged**